streamwebs_frontend/streamwebs/tests/models/test_water_quality.py

```diff
@@ -0,0 +1,124 @@
+from django.apps import apps
+from itertools import chain
+from django.core.exceptions import ValidationError
+
+from streamwebs.models import Site, Water_Quality, validate_wq_site
+
+
+class WaterQualityTestCase(TestCase):
+
+    def setUp(self):
+        self.expected_fields = {
+            'site_name': models.ForeignKey,
+            'date': models.DateField,
+            # 'time': models.TimeField,
+            'school': models.CharField,
+            'teacher': models.CharField,
+            'latitude': models.DecimalField,
+            'longitude': models.DecimalField,
+            'fish_present': models.BooleanField,
+            'live_fish': models.PositiveSmallIntegerField,
+            'dead_fish': models.PositiveSmallIntegerField,
+            'water_temp': models.DecimalField,
+            'air_temp': models.DecimalField,
+            'dissolved_O2': models.DecimalField,
+            'pH': models.DecimalField,
+            'turbidity': models.DecimalField,
+            'salinity': models.DecimalField,
+            'conductivity': models.DecimalField,
+            'total_solids': models.DecimalField,
+            'bod': models.DecimalField,
+            'ammonia': models.DecimalField,
+            'nitrite': models.DecimalField,
+            'nitrate': models.DecimalField,
+            'phosphates': models.DecimalField,
+            'fecal_coliform': models.DecimalField,
+            'notes': models.TextField,
+            'id': models.AutoField
+        }
+
+        # Where does 'Vernier' and 'Manual' fit into the model?
+        # Celcius and Fahrenheit?
+        # Sample number??
+
+        self.optional_fields = {
+            'conductivity',
+            'total_solids',
+            'bod',
+            'ammonia',
+            'nitrite',
+            'nitrate',
+            'phosphate',
+            'fecal_coliform',
+            'notes',
+        }
+
+    def test_fields_exist(self):
+        model = apps.get_model('streamwebs', 'water_quality')
+        for field, field_type in self.expected_field.items():
+            self.asserEqual(
+                field_type, type(model._meta.get_field(field)))
+
+    def test_no_extra_fields(self):
+        fields = list(set(chain.from_iterable(
+            (field.name, field.attname) if hasattr(field, 'attname') else
+            (field.name,) for field in Water_Quality._meta.get_fields()
+            if not (field.many_to_one and field.related_model is None)
+        )))
+        self.assertEqual(sorted(fields), sorted(self.expected_fields.keys()))
+
+    def test_optional_fields(self):
+        apps.get_model('streamwebs', 'site')
+        for field in self.optional_fields:
+            self.assertEqual(
+                Water_Quality._meta.get_field(field).blank, True)
+
+    def test_datasheet_OneToOne(self):
+        site = Site.objects.create('test', 'some_type', 'some_slug')
+        waterq = Water_Quality.objects.create(site=site, site_name='test',
```

Comment:
Does the model have both a foreign key 'site' and also a string site name? Isn't the name part of the site object?